src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,507 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-            } else if (value === "cdrom" && this.state.busType === "virtio") {
-                // use onValueChange instead of setState in order to perform subsequent state change logic
-                // According to https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms (section about 'target'),
-                // scsi is the default option for libvirt in this case too
-                this.onValueChanged("busType", "scsi");
-            }
+    const onValueChanged = (key, value) => {
+        switch (key) {
+        case 'storagePoolName': {
+            const currentPool = storagePools.find(pool => pool.name === value && pool.connectionName === vm.connectionName);
+            setDiskParams(diskParams => ({ ...diskParams, storagePool: currentPool, existingVolumeName: getDefaultVolumeName(currentPool, vm) }));
+            break;
+        }
+        case 'mode': {
+            setMode(value);
             break;
         }
         default:
-            this.setState({ [key]: value });
+            setDiskParams(diskParams => ({ ...diskParams, [key]: value }));
         }
+    };
+
+    const dialogErrorSet = (text, detail) => {
+        setDialogError(text);
+        setDialogErrorDetail(detail);
+    };
+
+    let defaultBody;
+    if (dialogLoading) {
+        defaultBody = (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    } else if (isMediaInsertion) {
+        defaultBody = (
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
+                <FormGroup fieldId={`${idPrefix}-source`}
+                           id={`${idPrefix}-source-group`}
+                           label={_("Source")} isInline hasNoPaddingTop>
+                    <Radio id={`${idPrefix}-custompath`}
+                           name="source"
+                           label={_("Custom path")}
+                           isChecked={mode === CUSTOM_PATH}
+                           onChange={() => onValueChanged('mode', CUSTOM_PATH)} />
+                    <Radio id={`${idPrefix}-useexisting`}
+                           name="source"
+                           label={_("Use existing")}
+                           isChecked={mode === USE_EXISTING}
+                           onChange={() => onValueChanged('mode', USE_EXISTING)} />
+                </FormGroup>
+                {mode === USE_EXISTING && (
+                    <UseExistingDisk idPrefix={`${idPrefix}-existing`}
+                                     onValueChanged={onValueChanged}
+                                     storagePoolName={storagePoolName}
+                                     existingVolumeName={diskParams.existingVolumeName}
+                                     validationFailed={validationFailed}
+                                     vmStoragePools={storagePools}
+                                     vms={vms}
+                                     vm={vm} />
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231113-155652-97d90b8c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)